packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2501,5 +2507,406 @@ describe("Runtime", () => {
+
+			it("accurately reports amount of sent and lost signals when roundtrip tracked signal is dropped", () => {
+				// Send 50 signals and drop 10
+				sendSignals(50);
+				dropSignals(10);
+				processSubmittedSignals(40);
+
+				// Send 60 signals and drop 15 (including roundtrip tracked signal)
+				sendSignals(60);
+				processSubmittedSignals(40);
+				dropSignals(15); // Drop roundtrip tracked signal
+
+				// Since roundtrip signal is lost, we don't expect to see SignalLatency telemetry for the first 100 signals
+				processSubmittedSignals(5);
+
+				// Send 100 signals and drop 1
+				sendSignals(100);
+				dropSignals(1);
+
+				// Here we should detect that 200 signals have been sent and 26 signals were lost
+				processSubmittedSignals(99);
+
+				// Check SignalLatency logs amount of sent and lost signals
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsSent: 201,
+							signalsLost: 26,
+						},
+					],
+					"SignalLatency telemetry should log absolute lost signal count for each batch of 100 signals",
+				);
+			});
+
+			it("accurately reports amount of sent and lost signals when rapid fire more than 100+ signals", () => {
+				sendSignals(1);
+				processSubmittedSignals(1);
+
+				sendSignals(101);
+				dropSignals(10);
+				processSubmittedSignals(91);
+
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsSent: 100,
+							signalsLost: 10,
+						},
+					],
+					"SignalLatency telemetry should log correct amount of sent and lost signals",
+				);
+			});
+
+			it("should log out of order signal in between signal latency events", () => {
+				// Send 1st signal and process it to prime the system
+				sendSignals(1);
+				processSubmittedSignals(1);
+				// Send 150 signals and temporarily lose 1
+				sendSignals(150); //            150 outstanding including 1 tracked signal (#101); max #151
+				processSubmittedSignals(95); //  55 outstanding including 1 tracked signal (#101)
+				dropSignals(1); //               54 outstanding including 1 tracked signal (#101)
+				processSubmittedSignals(14); //  40 outstanding; none tracked
+				processDroppedSignals(1); //     40 outstanding; none tracked *out of order signal*
+				processSubmittedSignals(40); //   0 outstanding; none tracked
+
+				// Send 60 signals including tracked signal
+				sendSignals(60); //              60 outstanding including 1 tracked signal (#201); max #211
+				processSubmittedSignals(60); //    0 outstanding; none tracked
+
+				// Check SignalLatency logs amount of sent and lost signals
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsSent: 100,
+							signalsLost: 1,
+						},
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsSent: 100,
+							signalsLost: 0,
+						},
```

Comment:
We should comment on the signalLost count in the second Latency event.
Could we let this be negative to account for the out of order signal (that used to be lost)?
Do we want the queries for lost signals to have to count SignalOutOfOrder events?
Should we bother with putting out of order counts in?

Reply:
I did think of the question of 'negative signalsLost' counts. The benefits I noticed was that we could have a more accurate representation of how many signals are lost when looking at the sent:lost ratio from summing SIgnalLatency events (since you're counting out of order signal events that fall outside of the latency window). The tradeoff is that individual SignalLatency events are harder to reason about since it isn't obvious what -2 signalsLost means immediately. 

Having an out of order count is probably the best solution. This is something I think would be nice to include.

Reply:
I too think out of order count could improve telemetry.